File: weather.py
#!/usr/bin/env python
# CoffeeOutsideBot
# Copyright 2016-2017, David Crosby
# BSD 2-clause license
#
# TODO - automate Cyclepalooza event creation

import os
import json
import logging
from ConfigParser import SafeConfigParser
import openweathermapy.core as owm

class Forecast:
    def __init__(self, config_file='./cb_config.ini'):
        parser = SafeConfigParser()
        parser.read(config_file)

        self.owm_config = {}
        # yuck
        for section in ['openweathermap']:
            if parser.has_section(section):
                if section == 'openweathermap':
                    for option in parser.options('openweathermap'):
                        self.owm_config[option] = parser.get('openweathermap', option)

        self.forecast = self.get_forecast()

    def get_forecast(self):
        # Use Celsius
        # TODO remove hardcode for Calgary city id
        data = owm.get_forecast_daily(5913490, 3, **self.owm_config)
        logging.debug(data[2])
        return data[2]

    def temperature(self):
        return self.forecast['temp']['morn']

    def humidity(self):
        return self.forecast['humidity']

if __name__ == '__main__':
    logging.basicConfig(level=logging.DEBUG)
    ee = Forecast()
    print("forecast temperature:", ee.temperature())
    print("forecast humidity:", ee.humidity())

File: weather.go
package main

// TODO - print forecast datetime, make sure it's accurate!!

import (
	"log"

	owm "github.com/briandowns/openweathermap"
	"gopkg.in/ini.v1"
)

type WeatherService struct {
	config_file string
	city_id     int
	api_key     string
}

type forecast struct {
	temp     int
	humidity int
}

func (ws *WeatherService) load_config() {
	cfg, err := ini.Load("cb_config.ini")
	if err != nil {
		log.Fatalf("Fail to read file: %v", err)
	}

	ws.city_id, _ = cfg.Section("openweathermap").Key("cty_id").Int()
	ws.api_key = cfg.Section("openweathermap").Key("appid").String()

}

func (ws WeatherService) get_forecast() (f forecast) {
	ws.load_config()
	w, err := owm.NewForecast("5", "C", "en", ws.api_key)
	if err != nil {
		log.Fatalln(err)
	}

	w.DailyByID(ws.city_id, 3)
	forecast_obj := w.ForecastWeatherJson.(*owm.Forecast5WeatherData)

	// split out details
	var fc forecast
	fc.temp = int(forecast_obj.List[2].Main.Temp)
	fc.humidity = forecast_obj.List[2].Main.Humidity

	log.Printf("Weather temp %v", fc.temp)
	log.Printf("Weather humidity %v", fc.humidity)

	return fc
}
